test/src/test/java/hudson/tasks/ArtifactArchiverTest.java

```diff
@@ -260,6 +274,126 @@ public void testDefaultExcludesOff() throws Exception {
+
+        j.buildAndAssertSuccess(p);
+        VirtualFile artifacts = p.getBuildByNumber(1).getArtifactManager().root();
+        assertTrue("file was archived", artifacts.child("target").child("file").exists());
+    }
+
+    @Test
+    @Issue("JENKINS-12379")
+    public void testBasePathUp() throws Exception {
+        FreeStyleProject p = j.createFreeStyleProject();
+
+        p.getBuildersList().replaceBy(Collections.singleton(new CreateFilesForBasePathTest()));
+
+        ArtifactArchiver archiver = new ArtifactArchiver("**");
+        archiver.setBasePath("..");
+        p.getPublishersList().replaceBy(Collections.singleton(archiver));
+
+        assertEquals(Result.FAILURE, build(p));
+        assertArrayEquals("artifacts list empty", new VirtualFile[0], p.getBuildByNumber(1).getArtifactManager().root().list());
+    }
+
+    @Test
+    @Issue("JENKINS-12379")
+    public void testBasePathInOutOut() throws Exception {
+        FreeStyleProject p = j.createFreeStyleProject();
+
+        p.getBuildersList().replaceBy(Collections.singleton(new CreateFilesForBasePathTest()));
+
+        ArtifactArchiver archiver = new ArtifactArchiver("**");
+        archiver.setBasePath("foo/../..");
+        p.getPublishersList().replaceBy(Collections.singleton(archiver));
+
+        assertEquals("build failed", Result.FAILURE, build(p));
+        assertArrayEquals("artifacts list empty", new VirtualFile[0], p.getBuildByNumber(1).getArtifactManager().root().list());
+    }
+
+    @Test
+    @Issue("JENKINS-12379")
+    public void testBasePathAbsolute() throws Exception {
+        FreeStyleProject p = j.createFreeStyleProject();
+
+        p.getBuildersList().replaceBy(Collections.singleton(new CreateFilesForBasePathTest()));
+
+        ArtifactArchiver archiver = new ArtifactArchiver("**");
+        // TODO absolute path on Windows?
+        archiver.setBasePath("/");
+        p.getPublishersList().replaceBy(Collections.singleton(archiver));
+
+        assertEquals("build failed", Result.FAILURE, build(p));
+        assertArrayEquals("artifacts list empty", new VirtualFile[0], p.getBuildByNumber(1).getArtifactManager().root().list());
+    }
+
+    @Test
+    @Issue("JENKINS-12379")
+    public void testBasePathValidation() throws Exception {
+        ArtifactArchiver.DescriptorImpl desc = (ArtifactArchiver.DescriptorImpl)j.jenkins.getDescriptorOrDie(ArtifactArchiver.class);
+
+        FreeStyleProject p = j.createFreeStyleProject();
+        assertEquals("no workspace", FormValidation.Kind.OK, desc.doCheckBasePath(p, "foo").kind);
+        assertEquals("explicit same directory", FormValidation.Kind.OK, desc.doCheckBasePath(p, ".").kind);
+        assertEquals("relative path breakout without workspace", FormValidation.Kind.ERROR, desc.doCheckBasePath(p, "..").kind);
+        assertEquals("absolute path breakout without workspace", FormValidation.Kind.ERROR, desc.doCheckBasePath(p, "/").kind);
```

Comment:
Windows? I guess it is OK since the master could be on Linux. But then you also need to check `C:\`, in case the master is on Linux and the agent is on Windows.